rclpy/rclpy/node.py

```diff
@@ -415,21 +422,30 @@ def get_parameter(self, name: str) -> Parameter:
             # If undeclared parameters are allowed, the parameter might be in the initial set.
             # If that's the case, first set and then return.
             if name in self._initial_parameters:
-                self._parameters.update({name: self._initial_parameters[name]})
-                self._descriptors.update({name: ParameterDescriptor()})
+                self._update_initial_parameter(name)
                 return self._parameters[name]
             return Parameter(name, Parameter.Type.NOT_SET, None)
         else:
             raise ParameterNotDeclaredException(name)
 
+    def _update_initial_parameter(self, name: str):
+        """Move initial parameter to parameter list."""
+        self._parameters.update({name: self._initial_parameters[name]})
+        self._descriptors.update({name: ParameterDescriptor()})
+        del self._initial_parameters[name]
+
     def get_parameter_or(
             self, name: str, alternative_value: Optional[Parameter] = None) -> Parameter:
         """
         Get a parameter or the alternative value.
 
+        If the parameter was not declared, if undeclared parameters are allowed and the parameter
+        is one of the initial parameters, it will be implicitly declared with a default descriptor
+        and returned.
```

Comment:
Same for this one.

Reply:
Addressed in https://github.com/ros2/rclpy/pull/339/commits/9227ad2e77040f1da47f431d0695f72dab84ec6b; waiting for CI now.